spark/dl/src/main/scala/com/intel/analytics/bigdl/dataset/Transformer.scala

```diff
@@ -206,31 +226,33 @@ class SampleToBatch[T: ClassTag]
           var featureIndex = 0
           var batchLength = 1
           while (i < batchSize && prev.hasNext) {
-            val sample = prev.next()
-            require(sample.feature().isContiguous() && sample.label().isContiguous(),
+            val s = prev.next()
+            require(s.isInstanceOf[TensorTensorSample[T]])
+            val sample = s.asInstanceOf[TensorTensorSample[T]]
+            require(sample.feature.isContiguous() && sample.label.isContiguous(),
```

Comment:
To avoid user code change, rename these two variables as featureTensor and labelTensor, and add a default (or deprecated) method feature() and label() in Sample to return these two tensors.